### PR TITLE
Adds migration docs for @amplitude/analytics-browser

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
# Dev Docs PR

## Description

Adds migration guide for moving from `amplitude-js` to `@amplitude/analytics-browser`

## Deadline

When do these changes need to be live on the site?

N/A

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [x] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

File: docs/data/sdks/typescript-browser/migration.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+To migrate to `@amplitude/analytics-browser`, the dependency and instrumentation must be updated.
+
+### Terminology
+
+* `amplitude-js`: Legacy Browser SDK
+* `@amplitude/analytics-browser`: New Browser SDK
+
+## Dependency
+
+For snippet installation, update your project"s snippet loader. The new snippet loader can be found here: [https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader)
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:17]: nits: project''s -> project's

File: docs/data/sdks/typescript-browser/migration.md [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+  }
+  ```
+
+## Instrumentation
+
+The new Browser SDK offers a new API to instrument events. To move forward with the migration, the following API calls must be changed.
+
+### Initialization
+
+Similar to all other calls, `getInstance()` has been removed. To initialize the SDK, call `init()`, with the same parameters. However, `config` comes in a different shape. See [Configuration](#configuration).
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:45]: Do we need to mention, that we don't support multiple instants for now?

File: docs/data/sdks/typescript-browser/migration.md [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+=== "@amplitude/analytics-browser"
+  ```typescript
+  import * as amplitude from "amplitude-js"
+
+  amplitude.init(API_KEY, OPTIONAL_USER_ID, config)
+  ```
+
+### Configuration
+
+The new Browser SDK configuration comes in a different shape. These were updated to have more consistency across other runtimes. In addition, some configurations are no unsupported for various reasons like, simplification, adopted as default hehavior, etc.
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:63]: typo "hehavior" -> behavior?

File: docs/data/sdks/typescript-browser/migration.md [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+|`config.headers`|NOT SUPPORTED|
+|`config.serverZone`|`config.serverZone`|
+|`config.useDynamicConfig`|NOT SUPPORTED|
+|`config.serverZoneBasedApi`|NOT SUPPORTED|
+|`config.sessionId`|`config.sessionId`|
+|`config.partnerId`|`config.partnerId`|
+
+### Tracking events
+
+The APIs to track events the most commonly used. `@amplitude/analytics-browser` offered a variety of `logEvent` APIs like `logEventWithTimestamp`, `logEventWithGroups` to override specific properties in the event payload. Amplitude has simplified all of these variations into a unified `track` API in `@amplitude/analytics-browser`.
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:119]: typo: `a unified` -> 'an unified'

File: docs/data/sdks/typescript-browser/migration.md [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+=== "amplitude-js"
+  ```javascript
+  const deviceId = "1"
+  amplitude.getInstance().setDeviceId(deviceId)
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const deviceId = "1"
+  amplitude.getDeviceId(deviceId)
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:267]: getDeviceId -> setDeviceId

File: docs/data/sdks/typescript-browser/migration.md [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+=== "@amplitude/analytics-browser"
+  ```typescript
+  amplitude.identify(
+    new amplitude.Identify().identify.clearAll()
+  )
+  ```
+
+#### `setUserProperties()`
+
+The `clearUserProperties` API has been removed; however the same funcationality can be achieved by using a unified `identify` API.
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:304]: `clearUserProperties ` -> `setUserProperties `

File: docs/data/sdks/typescript-browser/migration.md [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+  const revenue = new amplitude.Revenue()
+  revenue
+    .setProductId("productId")
+    .setPrice(10)
+  amplitude.revenue(revenue)
+  ```
+
+### Patterns
+
+#### Plugins
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:387]: Do we need to mention how to add a plugin,  and also add the destination plugin example?

File: docs/data/sdks/typescript-browser/migration.md [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+    execute(event: Event): Promise<Event> {
+      event.library = 'my-library-name/1.0.0'
+      return Promise.resolve(event)
+    }
+  }
+  ```
+
+#### Defer initialization
+
+To defer initialization in `amplitude-js`, init must be called with `config.deferInitialization` set to true, and eventually calling `enableTracking()` to formalize initialization and send all enqueued events.
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/migration.md:420]: I think we should follow the same pattern. One paragraph to mention the issue. and then one code snippet with 2 language.

File: docs/data/sdks/typescript-browser/migration.md [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+=== "amplitude-js"
+  ```json
+  {
+    "dependencies": {
+      "amplitude-js": "8"
+    }
+  }
+  ```
+
+=== "amplitude-js"
```
Comment (bgiori) [inline docs/data/sdks/typescript-browser/migration.md:30]: this should be "@amplitude/analytics-browser" ?

File: docs/data/sdks/typescript-browser/migration.md [hunk 10]
```diff
@@ -0,0 +1,497 @@
...
+  ```javascript
+  import amplitude from "amplitude-js"
+
+  amplitude.getInstance().init(API_KEY, OPTIONAL_USER_ID, config)
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  import * as amplitude from "amplitude-js"
```
Comment (bgiori) [inline docs/data/sdks/typescript-browser/migration.md:56]: import should be "@amplitude/analytics-browser" ?

File: docs/data/sdks/typescript-browser/migration.md [hunk 11]
```diff
@@ -0,0 +1,497 @@
...
+|`config.headers`|NOT SUPPORTED|
+|`config.serverZone`|`config.serverZone`|
+|`config.useDynamicConfig`|NOT SUPPORTED|
+|`config.serverZoneBasedApi`|NOT SUPPORTED|
+|`config.sessionId`|`config.sessionId`|
+|`config.partnerId`|`config.partnerId`|
+
+### Tracking events
+
+The APIs to track events the most commonly used. `@amplitude/analytics-browser` offered a variety of `logEvent` APIs like `logEventWithTimestamp`, `logEventWithGroups` to override specific properties in the event payload. Amplitude has simplified all of these variations into a unified `track` API in `@amplitude/analytics-browser`.
```
Comment (bgiori) [inline docs/data/sdks/typescript-browser/migration.md:119]: "The APIs to track events the most commonly used." 

This is superfluous.

File: docs/data/sdks/typescript-browser/migration.md [hunk 12]
```diff
@@ -0,0 +1,497 @@
...
+    () => {
+      amplitude.setTransport('beacon') // Optional. Sets https transport to use `sendBeacon` API
+      amplitude.flush()
+    },
+  );
+  ```
+
+#### Callback
+
+For `amplitude-js`, two separate callback functions are passed for success and error. With `@amplitude/analytics-browser` supporting Promises (and async/await), the asynchronous methods like `track()`, `identify()`, `groupIdentify` return a custom promise interface.
```
Comment (bgiori) [inline docs/data/sdks/typescript-browser/migration.md:486]: include `()` after `groupIdentify` for consistency.

**Review by bgiori — APPROVED** [review]
Nice, this should help most folks looking to switch 💯 

Approved pending changes.

File: docs/data/sdks/typescript-browser/migration.md [hunk 13]
```diff
@@ -0,0 +1,497 @@
+---
+title: Migration Guide
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:2]: Can you be a little more clear with the title, so someone searching this would be able to tell what it's for?

File: docs/data/sdks/typescript-browser/migration.md [hunk 14]
```diff
@@ -0,0 +1,497 @@
...
+description: This is a guide on migrating from amplitude-js
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:3]: this is the page's meta text, so can you add a little more detail here? for example: 

Use this guide to easily migrate from Amplitude's legacy browser SDK (amplitude-js) to the new SDK (@amplitude/analytics-browser).

File: docs/data/sdks/typescript-browser/migration.md [hunk 15]
```diff
@@ -0,0 +1,497 @@
...
+---
+
+Amplitude has released a new version of Browser SDK (`@amplitude/analytics-browser`), which features a plugin architecture, built-in type definition and broader support for front-end frameworks. The new version is not backwards compatible with `amplitude-js`.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:6]:
```suggestion
The new version of Amplitude's Browser SDK (`@amplitude/analytics-browser`) features a plugin architecture, built-in type definition and broader support for front-end frameworks. The new version is not backwards compatible with `amplitude-js`. 
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 16]
```diff
@@ -0,0 +1,497 @@
...
+To migrate to `@amplitude/analytics-browser`, the dependency and instrumentation must be updated.
+
+### Terminology
+
+* `amplitude-js`: Legacy Browser SDK
+* `@amplitude/analytics-browser`: New Browser SDK
+
+## Dependency
+
+For snippet installation, update your project"s snippet loader. The new snippet loader can be found here: [https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader)
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:17]:
```suggestion
For snippet installation, update your project's [snippet loader](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader).
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 17]
```diff
@@ -0,0 +1,497 @@
...
+  {
+    "dependencies": {
+      "@amplitude/analytics-browser": "1"
+    }
+  }
+  ```
+
+## Instrumentation
+
+The new Browser SDK offers a new API to instrument events. To move forward with the migration, the following API calls must be changed.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:41]:
```suggestion
The new Browser SDK offers an API to instrument events. To move forward with the migration, the following API calls must be changed.
```

Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:41]: This is kind of unclear: "the following API calls must be changed". Is there supposed to be a list here, or is it just referring to the calls in the following sections? 

I'd reword to something like: 
To migrate to the new SDK, you need to update a few calls. The following sections detail which calls need to be changed.

File: docs/data/sdks/typescript-browser/migration.md [hunk 18]
```diff
@@ -0,0 +1,497 @@
...
+=== "@amplitude/analytics-browser"
+  ```typescript
+  import * as amplitude from "amplitude-js"
+
+  amplitude.init(API_KEY, OPTIONAL_USER_ID, config)
+  ```
+
+### Configuration
+
+The new Browser SDK configuration comes in a different shape. These were updated to have more consistency across other runtimes. In addition, some configurations are no unsupported for various reasons like, simplification, adopted as default hehavior, etc.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:63]: What does "some configurations are no unsupported...." mean here?

File: docs/data/sdks/typescript-browser/migration.md (hunk 11 above)
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:119]:
```suggestion
The legacy Browser SDK offered a variety of `logEvent` APIs like `logEventWithTimestamp`, `logEventWithGroups` to override specific properties in the event payload. Amplitude has simplified all of these variations into a unified `track` API in `@amplitude/analytics-browser`.
```

I made a guess at the meaning of this.

File: docs/data/sdks/typescript-browser/migration.md [hunk 19]
```diff
@@ -0,0 +1,497 @@
...
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  amplitude.flush()
+  ```
+
+### Setting user properties
+
+The APIs for setting user properties remain familiar with the exception of the removal of `getInstance()`. Here are code snippets to migrate APIs for user properties.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:236]:
```suggestion
The APIs for setting user properties are the same, with the exception of the removal of `getInstance()`. Here are code snippets to migrate APIs for user properties.
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 20]
```diff
@@ -0,0 +1,497 @@
...
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const sessionId = Date.now()
+  amplitude.setSessionId(sessionId)
+  ```
+
+#### `clearUserProperties()`
+
+The `clearUserProperties` API has been removed; however the same funcationality can be achieved by using a unified `identify` API.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:288]:
```suggestion
The `clearUserProperties` API has been removed, but you can now use the unified `identify` API to remove user properties. 
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 21]
```diff
@@ -0,0 +1,497 @@
...
+=== "@amplitude/analytics-browser"
+  ```typescript
+  amplitude.identify(
+    new amplitude.Identify().identify.clearAll()
+  )
+  ```
+
+#### `setUserProperties()`
+
+The `clearUserProperties` API has been removed; however the same funcationality can be achieved by using a unified `identify` API.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:304]:
```suggestion
The `setUserProperties` API has been removed, but you can now use the unified `identify` API to add user properties. 
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 22]
```diff
@@ -0,0 +1,497 @@
...
+  const identify = new amplitude.Identify()
+  identify
+    .set("membership", "paid")
+    .set("payment", "bank")
+  amplitude.identify(identify)
+  ```
+
+#### `identify()`
+
+Making an identify call can be done on `amplitude` without calling `getInstance()`.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:325]:
```suggestion
You can now make an identify call on `amplitude` without calling `getInstance()`.
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 23]
```diff
@@ -0,0 +1,497 @@
...
+  const identify = new amplitude.Identify()
+  identify.set("membership", "paid")
+  amplitude.identify(identify)
+  ```
+
+### Setting group properties
+
+### `groupIdentify()`
+
+Making an identify call can be done on `amplitude` without calling `getInstance()`.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:345]:
```suggestion
You can now make an identify call on `amplitude` without calling `getInstance()`.
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 24]
```diff
@@ -0,0 +1,497 @@
...
+  const identify = new amplitude.Identify()
+  identify.set("membership", "paid")
+  amplitude.groupIdentify(identify)
+  ```
+
+### Tracking revenue
+
+#### `logRevenueV2()`
+
+Tracking revenue can be done using `revenue()` API on `amplitude` without calling `getInstance()`.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:365]:
```suggestion
Track revenue using `revenue()` API on `amplitude` without calling `getInstance()`.
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 25]
```diff
@@ -0,0 +1,497 @@
...
+    .setProductId("productId")
+    .setPrice(10)
+  amplitude.revenue(revenue)
+  ```
+
+### Patterns
+
+#### Plugins
+
+The configs `config.language`, `config.library`, `config.platform` were added to `amplitude-js` to allow modification of event payloads for these specific fields. While `@amplitude/analytics-browser` do not support these configurations anymore, the new Browser SDK allows for plugins to be added and enrich event payloads.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:389]:
```suggestion
The configs `config.language`, `config.library`, `config.platform` were available in `amplitude-js` to allow modification of event payloads for these specific fields. Although `@amplitude/analytics-browser` doesn't support these configurations, you can add plugins to the new Browser SDK to enrich event payloads.
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 26]
```diff
@@ -0,0 +1,497 @@
...
+=== "amplitude-js"
+  ```javascript
+  amplitude.getInstance().init(API_KEY, OPTIONAL_USER_ID, {
+    onExitPage: () => {
+      amplitude.sendEvents()
+    },
+  })
+  ```
+
+For `@amplitude/analytics-browser`, Amplitude recommends adding your own event listerer for `pagehide` event.
```
Comment (caseyamp) [inline docs/data/sdks/typescript-browser/migration.md:472]:
```suggestion
For `@amplitude/analytics-browser`, Amplitude recommends adding your own event listener for `pagehide` event.
```

**Review by caseyamp — CHANGES_REQUESTED** [review]
Howdy, I had a lot of questions and reworded some things to use a more active voice. LMK if you have any questions.

File: docs/data/sdks/typescript-browser/migration.md [hunk 27]
```diff
@@ -0,0 +1,497 @@
...
+      "amplitude-js": "8"
+    }
+  }
+  ```
+
+=== "amplitude-js"
+  ```json
+  {
+    "dependencies": {
+      "@amplitude/analytics-browser": "1"
```
Comment (justin-fiedler) [inline docs/data/sdks/typescript-browser/migration.md:34]: I think we should promote semantic versioning e.g. "^1.0.0"

File: docs/data/sdks/typescript-browser/migration.md [hunk 28]
```diff
@@ -0,0 +1,497 @@
...
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const eventType = "Button Clicked"
+  const eventProperties = {
+    type: "primary",
+  }
+  amplitude.track(
+    eventType,
+    eventProperties,
+  ).promise
```
Comment (justin-fiedler) [inline docs/data/sdks/typescript-browser/migration.md:146]: Maybe call out the promise in a separate section. I think people could get confused here and think they need to use the promise.

**Review by justin-fiedler — COMMENTED** [review]
Looks good @kevinpagtakhan. I think it might help if the code snippets could be aligned side by side quick comparison rather than over/under.

File: docs/data/sdks/typescript-browser/migration.md (hunk 1 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:17]:
```suggestion
For snippet installation, update your project's snippet loader. The new snippet loader can be found here: [https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader)
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 3 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:63]:
```suggestion
The new Browser SDK configuration comes in a different shape. These were updated to have more consistency across other runtimes. In addition, some configurations are no unsupported for various reasons like, simplification, adopted as default behavior, etc.
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 9 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:30]:
```suggestion
=== "@amplitude/analytics-browser"
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 27 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:34]:
```suggestion
      "@amplitude/analytics-browser": "^1.0.0"
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 29]
```diff
@@ -0,0 +1,497 @@
...
+
+For snippet installation, update your project"s snippet loader. The new snippet loader can be found here: [https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/analytics-browser#using-script-loader)
+
+For Node projects, update your dependency list in package.json.
+
+=== "amplitude-js"
+  ```json
+  {
+    "dependencies": {
+      "amplitude-js": "8"
```
Comment (kevinpagtakhan) [inline docs/data/sdks/typescript-browser/migration.md:25]:
```suggestion
      "amplitude-js": "^8.0.0"
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 10 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:56]:
```suggestion
  import * as amplitude from "@amplitude/analytics-browser"
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 5 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:267]:
```suggestion
  amplitude.setDeviceId(deviceId)
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 12 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:486]:
```suggestion
For `amplitude-js`, two separate callback functions are passed for success and error. With `@amplitude/analytics-browser` supporting Promises (and async/await), the asynchronous methods like `track()`, `identify()`, `groupIdentify()` return a custom promise interface.
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 13 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:2]:
```suggestion
title: Browser SDK Migration Guide
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 14 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:3]:
```suggestion
description: Use this guide to easily migrate from Amplitude's legacy browser SDK (amplitude-js) to the new SDK (@amplitude/analytics-browser).
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 2 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:45]: multi-instance support is scheduled to be worked on this week @yuhao900914

File: docs/data/sdks/typescript-browser/migration.md (hunk 28 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:146]:
```suggestion
  )
```

Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:146]: Removing `.promise` here. There's a dedicated section for callbacks :)

File: docs/data/sdks/typescript-browser/migration.md (hunk 17 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:41]:
```suggestion
The new Browser SDK offers an API to instrument events. To migrate to the new SDK, you need to update a few calls. The following sections detail which calls need to be changed.
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 30]
```diff
@@ -0,0 +1,497 @@
...
+    * execute() is called on each event instrumented
+    * example: client.track('New Event');
+    */
+    execute(event: Event): Promise<Event> {
+      event.library = 'my-library-name/1.0.0'
+      return Promise.resolve(event)
+    }
+  }
+  ```
+
```
Comment (kevinpagtakhan) [inline docs/data/sdks/typescript-browser/migration.md:417]:
```suggestion

To install your custom plugin, use `add()` with your custom plugin as parameter.

=== "@amplitude/analytics-browser"
  ```typescript
  import { add } from "@amplitude/analytics-browser"

  add(new LibraryModifierPlugin())
  ```
```

File: docs/data/sdks/typescript-browser/migration.md [hunk 31]
```diff
@@ -0,0 +1,504 @@
...
+### Patterns
+
+#### Plugins
+
+The configs `config.language`, `config.library`, `config.platform` were available in `amplitude-js` to allow modification of event payloads for these specific fields. Although `@amplitude/analytics-browser` doesn't support these configurations, you can add plugins to the new Browser SDK to enrich event payloads.
+
+=== "@amplitude/analytics-browser"
+  ```ts
+  import { BrowserConfig, EnrichmentPlugin, Event, PluginType } from '@amplitude/analytics-types'
```
Comment (kevinpagtakhan) [inline docs/data/sdks/typescript-browser/migration.md:393]:
```suggestion
  import { BrowserConfig, EnrichmentPlugin, Event, PluginType } from "@amplitude/analytics-types"
```

File: docs/data/sdks/typescript-browser/migration.md (hunk 7 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md:387]: Added

File: docs/data/sdks/typescript-browser/migration.md (hunk 4 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md]: I think since unified ends starts with a consonant _sound_, it should be `a`

File: docs/data/sdks/typescript-browser/migration.md (hunk 18 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-browser/migration.md]: Was a typo! Updated...

**Comment by kevinpagtakhan** [comment]
> Looks good @kevinpagtakhan. I think it might help if the code snippets could be aligned side by side quick comparison rather than over/under.

@justin-fiedler this would really be nice. but i think with limited width on the dev docs page, the code snippets maybe harder to read. on the page itself, we have side navs on both sides. any thoughts?

**Review by bgiori — APPROVED** [review]

**Review by qingzhuozhen — APPROVED** [review]

**Review by caseyamp — APPROVED** [review]
